src/NuGet.Core/NuGet.Protocol.Core.v3/HttpSource/HttpRetryHandler.cs

```diff
@@ -86,6 +86,8 @@ public HttpRetryHandler()
                             Strings.Http_RequestLog,
                             request.Method,
                             requestUri));
+                        
+                        response?.Dispose();
```

Comment:
Talked to @joelverhagen offline. I think it's worth adding comments why it's disposed here and not say in finally.